base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -40,17 +40,21 @@ pub const LOG_TARGET: &str = "c::val::transaction_validators";
 /// This function does NOT check that inputs come from the UTXO set
 pub struct TxInternalConsistencyValidator {
     factories: CryptoFactories,
+    bypass_range_proof_verification: bool,
 }
 
 impl TxInternalConsistencyValidator {
-    pub fn new(factories: CryptoFactories) -> Self {
-        Self { factories }
+    pub fn new(factories: CryptoFactories, bypass_range_proof_verification: bool) -> Self {
+        Self {
+            factories,
+            bypass_range_proof_verification,
+        }
     }
 }
 
 impl MempoolTransactionValidation for TxInternalConsistencyValidator {
     fn validate(&self, tx: &Transaction) -> Result<(), ValidationError> {
-        tx.validate_internal_consistency(&self.factories, None)
+        tx.validate_internal_consistency(self.bypass_range_proof_verification, &self.factories, None)
```

Comment:
I dont think the mempool should have this option, this leaves the node to propagate bad transactions to the rest of the network.

Reply:
Yeah I think I agree with this